src/UI/Components/TreeTable/RowReferenceContext.tsx

```diff
@@ -0,0 +1,12 @@
+import { createContext } from "react";
+
+interface TreeTableCell {
+  onClick(cellValue: string): void;
+}
+
+export const TreeTableCellContext = createContext<TreeTableCell>({
```

Comment:
I'm not sure I understand this part, why is it needed?

Reply:
It's a way to pass this data down the component tree, without having to pass the props explicitly at every level in between (https://en.reactjs.org/docs/context.html).
In this specific case this is useful, because we have the refs to the service entity rows in the parent component (the `DataList`), and we need to do an action with it (expand the row and scroll to it) deep in the component tree, at the cell level in the `TreeTable`. (We'll be able to use it for the inventory view as well).
The naming is not the best, I'm open to suggestions :slightly_smiling_face:

Reply:
the name is fine by me, I don't think I will find a better one.